agent/src/agentclient.rs

```diff
@@ -257,45 +257,133 @@ impl<T: APIServerClient> BrupopAgent<T> {
         Ok(())
     }
 
+    #[instrument(skip(self))]
+    async fn create_custom_resource_if_not_exist(&self) -> Result<()> {
```

Comment:
According to Sean's comment below, I think might be better to be consistent to use `shadow` instead of `custom resource`.
```suggestion
    async fn create_shadow_if_not_exist(&self) -> Result<()> {
```